ci.sh

```diff
@@ -23,32 +23,43 @@ function run_tests {
   fi
 
   ./miri test --locked
-  if ! [ -n "${MIRI_TEST_TARGET+exists}" ]; then
+  if [ -z "${MIRI_TEST_TARGET+exists}" ]; then
     # Only for host architecture: tests with MIR optimizations
     #FIXME: Only testing opt level 1 due to <https://github.com/rust-lang/rust/issues/77564>.
     MIRIFLAGS="-Z mir-opt-level=1" ./miri test --locked
   fi
+
+  if command -v python3; then
+    PYTHON=python3
+  else
+    PYTHON=python
+  fi
```

Comment:
If we still need such a hack, then why did you do the `PATH` thing for AppVeyor?

Please add a comment saying which platform still uses the old `python`. Once we use only GHA, can we just use `python3` unconditionally?

Reply:
Just a condition check. On Windows, there is always "python", not "python3" or "python2".

> why did you do the PATH thing for AppVeyor

On appveyor python isn't ready in PATH, we have to add python path to PATH.
The path appending already exists, just that python from mingw.

Reply:
> On Windows, there is always "python", not "python3" or "python2".

Okay, please add a comment explaining that.

> The path appending already exists, just that python from mingw.

I do not understand this sentence, there seems to be a verb missing (and maybe more)... what do you mean?

Reply:
> I do not understand this sentence

I mean this line already exists: the python path from mingw.
https://github.com/rust-lang/miri/blob/9d4c80e787f400e0ac36685b51460e52c1f31412/.appveyor.yml#L39

Reply:
(a) That line was nicely hidden in the platform that required it, which is better than having it in `ci.sh`, and
(b) given that this is a "python3.exe", does that not contradiction what you said above that there is no "python3" on Windows?

Reply:
b) that's mingw python. Python extracted from PythonTools of Windows components doesn't have python3.
a) either the problem solved in ci templates or in ci.sh. Which one you prefer ?

Reply:
> b) that's mingw python. Python extracted from PythonTools of Windows components doesn't have python3.

Ah makes sense, thanks.

> a) either the problem solved in ci templates or in ci.sh. Which one you prefer ?

I guess ci.sh works, now that you already did it.